src/shell/layout/mod.rs

```diff
@@ -126,10 +55,35 @@ pub fn is_dialog(window: &CosmicSurface) -> bool {
     false
 }
 
-pub fn has_floating_exception(window: &CosmicSurface) -> bool {
+#[derive(Debug, Clone)]
+pub struct TilingExceptions {
+    app_ids: RegexSet,
+    titles: RegexSet,
+}
+
+impl TilingExceptions {
+    pub fn new(config: &Config) -> Self {
+        let mut app_ids = Vec::new();
+        let mut titles = Vec::new();
+
+        for app in &config.cosmic_conf.tiling_exceptions {
+            for title in &app.titles {
+                app_ids.push(app.appid.clone());
+                titles.push(title.clone());
+            }
+        }
+
+        Self {
+            app_ids: RegexSet::new(app_ids).unwrap(),
+            titles: RegexSet::new(titles).unwrap(),
```

Comment:
Can this fail? Or is it safe to use `unwrap` here?

Reply:
After some more consideration I am pretty sure these would be good to check. The question is how to handle a potential failure.

Reply:
Maybe return an error in the TilingExceptions::new function, and log it when you call it.

Reply:
Let's start with that.

Reply:
I don't have a full picture of the code, but you should also make sure the log don't spam to much

Reply:
All of this happens in Shell init so it logs early and only once per start.